sympy/core/numbers.py

```diff
@@ -873,11 +873,6 @@ def __new__(cls, num, dps=None, precision=None):
             if isinstance(num[1], str):
                 # it's a hexadecimal (coming from a pickled object)
                 num = list(num)
-                # If we're loading an object pickled in Python 2 into
-                # Python 3, we may need to strip a tailing 'L' because
-                # of a shim for int on Python 3, see issue #13470.
-                if num[1].endswith('L'):
```

Comment:
I see no reason not to keep this intact.

Reply:
Reverted.